### PR TITLE
Edit profile

### PR DESCRIPTION
**Pull request by aulonashabani**

### CODE REVIEW

**Comment by michaelneu** [comment]
@mshoaib91 should be reviewing this, to keep our review circle up.

File: src/actions/biographyActions.ts [hunk 1]
```diff
@@ -54,15 +55,36 @@ const failUpdatingBiography = (error: string) : Action<string> => ({
...
+  };
+};
+
+export const addRealEstate = () => {
+  return async (dispatch: Dispatch) => {
+    dispatch(startUpdatingBiography());
+    dispatch(startSaving());
+
+    try {
+      const updatedBio = await api.createAccommodation();
```
Comment (tomasruizt) [inline src/actions/biographyActions.ts]: Lets keep a consistent naming as RealEstate

File: src/actions/imagesAction.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  return async (dispatch: Dispatch) => {
+    dispatch(startSaving());
+    await api.uploadProfileImage(image);
+
+    await fetchBiography()(dispatch);
+    await finishSaving()(dispatch);
+  };
+};
+
+export const uploadAccommodationImage = (accommodationId: string, image: File) => {
```
Comment (tomasruizt) [inline src/actions/imagesAction.ts]: Either they are all accommodations or all realEstates :D

File: src/api/client.ts [hunk 3]
```diff
@@ -81,6 +81,27 @@ export class ApiClient extends HttpClient {
...
 
+  public async uploadProfileImage(image: File) : Promise<any> {
+    const form = new FormData();
+
+    form.set("image", image);
+
+    return await this.post<any>(`${this.baseUrl}/images/profile`, form);
+  }
+
+  public async uploadAccommodationImage(accommodationId: string, image: File) : Promise<any> {
```
Comment (tomasruizt) [inline src/api/client.ts]: here as well

**Review by tomasruizt — CHANGES_REQUESTED** [review]

File: src/actions/imagesAction.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  return async (dispatch: Dispatch) => {
+    dispatch(startSaving());
+    await api.uploadAccommodationImage(accommodationId, image);
+
+    await fetchBiography()(dispatch);
```
Comment (m5khan) [inline src/actions/imagesAction.ts]: Why are we fetching biography in image upload action 🤦‍♂️

**Review by m5khan — APPROVED** [review]
In Michi I trust :D

**Pull request merged**